### PR TITLE
Dynamic worker pool task

### PR DESCRIPTION
**Pull request by Metalscreame**

### CODE REVIEW

File: concurrency/5/worker_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	for i := 0; i < 100; i++ {
+		select {
+		case tasksCh <- taskFunc:
+		case <-time.After(time.Second * 4):
+			t.Fatalf("task is not executed after 4 seconds")
+		}
+	}
+
+	//---------------
+	fmt.Println("Testing 'max load test case'")
```
Comment (arbrix) [inline concurrency/5/worker_test.go:33]: I've not found requirements that cover this test case could you please point me on it. 

For example, 100 tasks should be handled by a maximum number of workers within 1 sec.

File: concurrency/5/worker_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	time.Sleep(time.Second * 2)
+	currentWorkers := atomic.LoadInt32(wp.currentWorkerCount)
+	require.EqualValues(t, maxWorkersCount, currentWorkers,
+		"there should be max number of workers %v, but got %v", maxWorkersCount, currentWorkers)
+	fmt.Println("Successfully passed")
+	//---------------
+	fmt.Println("Testing 'load has gone' test case")
```
Comment (arbrix) [inline concurrency/5/worker_test.go:40]: the same for downscale, should we define cooldown period?

File: concurrency/5/worker_test.go (hunk 1 above)
Comment (Metalscreame) [reply concurrency/5/worker_test.go:33]: particularly this one is hard to define as it depends on the tasks itself

File: concurrency/5/worker_test.go (hunk 2 above)
Comment (Metalscreame) [reply concurrency/5/worker_test.go:40]: defined it in req

**Review by arbrix — APPROVED** [review]
Great work, thank you.

**Pull request merged**